lms/djangoapps/discussion_api/views.py

```diff
@@ -75,27 +76,41 @@ class CourseTopicsView(DeveloperErrorViewMixin, APIView):
     **Example Requests**:
 
         GET /api/discussion/v1/course_topics/course-v1:ExampleX+Subject101+2015
+            ?topic_id=a29d117faf43c7c59eda7717e722e9d8f4a47ea8
 
     **Response Values**:
+        if topic_id is not given:
+            * courseware_topics: The list of topic trees for courseware-linked
+              topics. Each item in the list includes:
 
-        * courseware_topics: The list of topic trees for courseware-linked
-          topics. Each item in the list includes:
+                * id: The id of the discussion topic (null for a topic that only
+                  has children but cannot contain threads itself).
 
-            * id: The id of the discussion topic (null for a topic that only
-              has children but cannot contain threads itself).
+                * name: The display name of the topic.
+
+                * children: A list of child subtrees of the same format.
+
+            * non_courseware_topics: The list of topic trees that are not linked to
+                  courseware. Items are of the same format as in courseware_topics.
+
+        if topic_id is given:
+            * id: The id of the discussion topic.
 
             * name: The display name of the topic.
 
             * children: A list of child subtrees of the same format.
-
-        * non_courseware_topics: The list of topic trees that are not linked to
-              courseware. Items are of the same format as in courseware_topics.
     """
```

Comment:
I'm not sure I like having one point endpoint return responses with such different formats depending on an optional parameter.  It feels like a normal list/detail pattern (list of topics vs. individual topic), where the list should show up at `/path/to/list/` and the the detail at `/path/to/list/{detail_id}/`.  Having the same endpoint handle both cases makes it less clear for client code to figure out what format to expect.  With the parameter, the top-level of the response is a topic object, and without it, it's a dictionary of two lists. 

I know @robrap and @nasthagiri were pushing to reuse the same endpoint, and I don't want to just be contradictory, so I'd love to hear further discussion on this point.